p4-16/spec/P4-16-draft-spec.mdk

```diff
@@ -4928,20 +4927,18 @@ table smac {
 }
 ~ End P4Example
 
-- The ```smac``` ```table``` can contain two types of actions, named ```Drop_action```
-  and ```Rewrite_mac```.
-- The ```Rewrite_smac``` action has one parameter, which is bound by
-  the control plane.
+- The entries in the ```smac``` ```table``` may contain two different actions: ```Drop_action``` and ```Rewrite_mac```.
+- The ```Rewrite_smac``` action has one parameter, ```sourceMac```, which is bound by the control plane.
 
-All actions in the list of actions of a table must have different
-names; e.g., the following program fragment is illegal:
+Each action in the list of actions for a table must have a distinct
+name---e.g., the following program fragment is illegal:
 
 ~ Begin P4Example
 action a() {}
 control c() {
     action a() {}
     // Illegal table: two actions with the same name
-    table t { actions = { a; .a; } }
```

Comment:
I see what this is doing, but the "." is subtle. I wonder if it's better just to leave it off...